perf_guide.md

```diff
@@ -67,17 +67,49 @@ addition is 2.7x slower than the one using vertical vector operations!
 
 ## Target-features
 
+Not all processors of a certain architecture will have SIMD processing units,
+and using a SIMD instruction when it's not supported will trigger a CPU exception.
+
+To allow building safe, portable programs, the Rust compiler will **not**, by default,
+generate any sort of vector instructions, unless it can statically determine
+they are supported (for example, on AMD64, SSE2 support is architecturally guaranteed).
+
 ### RUSTFLAGS
 
-TODO
+One of the easiest ways to benefit from SIMD is to allow the compiler
+to generate code using certain vector instruction extensions.
+
+The environment variable `RUSTFLAGS` can be used to pass options for code
+generation to the Rust compiler. These flags will affect **all** compiled crates.
+
+There are two flags which can be used to enable specific vector extensions:
+
+- `-C target-feature=<features>`: a comma-separated set of instruction extensions
+  to enable.
+  - **NOTE** (x86 specific): the `avx2` feature will _not_ enable `fma`, even though
```

Comment:
I would make this a general statement and say that target features must, in general, be enabled independently of each other, using `avx2` as an example, also showing the syntax. "For example, `-C target-feature=+avx2` does not enable the `fma` ... To enable both one has to write `-C target-feature=+avx2,+fma`.

Reply:
It is also worth mentioning that some target-features require others to be enabled. For example, to use `neon` on ARM targets without v7 in the base cpu, one has to enable both v7 and neon using `-C target-feature=+v7,+neon`.